tensorflow/python/keras/utils/vis_utils.py

```diff
@@ -124,6 +147,26 @@ def model_to_dot(model,
     dot.set('dpi', dpi)
     dot.set_node_defaults(shape='record')
 
+  if layer_range is None:
+    layer_range = [0, len(model.layers)]
+  
+  # check point for layer range whether its string or integer 
+  # layer range must be same dtype (mixed dtype is not allowed)
+  if all(map(lambda x: str(x).isdigit(), layer_range)):
```

Comment:
Why are we converting x to str(x) first? Shouldn't we just treat "1" as string value (layer name)?

Reply:
It's assumed to take as valid input if user passes `layer_range = [2, 2] or ["2", "2"]`, just some option to have for cases.

Reply:
I agree with @qlzh727 here. User can specify layer name as "2" and why should we convert that to index 2 ?
As of now, changed to only accept string and not integer.